test-infra/camel-test-infra-kafka/src/test/java/org/apache/camel/test/infra/kafka/services/ContainerLocalSingletonKafkaService.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.camel.test.infra.kafka.services;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.testcontainers.containers.KafkaContainer;
+import org.testcontainers.utility.DockerImageName;
+
+public class ContainerLocalSingletonKafkaService extends ContainerLocalKafkaService
+        implements ExtensionContext.Store.CloseableResource {
+
+    public ContainerLocalSingletonKafkaService(KafkaContainer container) {
+        super(container);
+    }
+
+    public ContainerLocalSingletonKafkaService() {
+        super();
+    }
+
+    @Override
+    public void beforeAll(ExtensionContext extensionContext) {
+        extensionContext.getRoot().getStore(ExtensionContext.Namespace.GLOBAL).getOrComputeIfAbsent("kafka", s -> {
+            super.initialize();
+            return this;
+        });
+    }
+
+    @Override
+    public void afterAll(ExtensionContext extensionContext) {
+        // no op
+    }
+
+    @Override
+    public void close() {
+        super.shutdown();
+    }
+
+    public static ContainerLocalSingletonKafkaService kafka3Container() {
+        KafkaContainer container
+                = new KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka:7.0.1")).withEmbeddedZookeeper();
```

Comment:
It would be nice, if we could move the name of the image in static field like this example https://github.com/apache/camel/blob/main/test-infra/camel-test-infra-cassandra/src/test/java/org/apache/camel/test/infra/cassandra/services/CassandraLocalContainerService.java#L30

Reply:
Hey @essobedo, not meaning to overrule you, but I am going to merge the code as is. I'll take responsibility of doing the change you suggested (I also talked w/ @johnpoth about it and it's fine for him). 

I want to apply the same idea to other services in the test-infra and I have time right now to focus on it.

Reply:
No worries, I only make suggestions, you are always free to follow them or not